File: scripts/circle_map_variants.R
suppressPackageStartupMessages({
  library(tidyverse)
})

args <- commandArgs(trailingOnly = TRUE)
variant_file <- args[1]
circle_file <- args[2]
output_file <- args[3]

variant_df <- read.delim(variant_file)
circle_df <- read.delim(circle_file, header = FALSE)

circle_variants_df <- data.frame()
for (row_num in 1:nrow(circle_df)) {
  row_chr <- circle_df[row_num, 1]
  row_start <- circle_df[row_num, 2]
  row_end <- circle_df[row_num, 3]
  new_variants <- filter(variant_df, CHROM == row_chr, POS > row_start, POS <= row_end)
  circle_variants_df <- rbind(circle_variants_df, new_variants)
}

circle_variants_df <- unique(circle_variants_df)
write_tsv(circle_variants_df, output_file)

File: scripts/circle_map_variants.py
import pandas as pd
import sys

variant_file = sys.argv[1]
circle_file = sys.argv[2]
output_file = sys.argv[3]

variant_df = pd.read_csv(variant_file, sep = "\t", header = 0, dtype=str)
variant_df['POS'] = variant_df['POS'].astype(int)
circle_df = pd.read_csv(circle_file, sep = "\t", header = None)

circle_variants_df = pd.DataFrame()
for row in circle_df.index:
    row_chr = circle_df.iloc[row, 0]
    row_start = circle_df.iloc[row, 1]
    row_end = circle_df.iloc[row, 2]
    new_variants_df = variant_df[(variant_df['CHROM'] == row_chr) & (variant_df['POS'] > row_start) & (variant_df['POS'] <= row_end)]
    circle_variants_df = pd.concat([circle_variants_df, new_variants_df])
    circle_variants_df = circle_variants_df.drop_duplicates()

circle_variants_df.to_csv(output_file, sep = "\t", header = True, index = None)